prometheus-to-sd/main.go

```diff
@@ -42,18 +43,36 @@ var (
 		"Comma-separated list of whitelisted metrics. If empty all metrics will be exported.")
 	apioverride = flag.String("api-override", "",
 		"The stackdriver API endpoint to override the default one used (which is prod).")
+	source = flags.Uris{}
 )
 
 func main() {
 	flag.Set("logtostderr", "true")
+	flag.Var(&source, "source", "source(s) to watch in [component]:http://host:port?whitelisted=a,b,c format")
```

Comment:
I think the best approach is to:
- make whitelisted optional
- introduce new flag "--auto-whitelist-metrics" that will make override the sources without whitelisted set (it would be still possible to use autodiscovery for only some sources)